### PR TITLE
Fix sonar warnings

### PR DESCRIPTION
**Pull request by osmandin**
https://jira.duraspace.org/browse/FCREPO-1336

### CODE REVIEW

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpResourceConverter.java [hunk 1]
```diff
@@ -449,4 +449,9 @@ private String txSegment() {
             return EMPTY;
         }
     }
+
+    @Override
```
Comment (awoods) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpResourceConverter.java]: Not supporting `hashCode()` seems like an a sign of a more fundamental issue.
But any ways, the related Sonar issue seems to be gone: http://sonar.fcrepo.org/drilldown/issues/1?&rule=findbugs%3AEQ_COMPARETO_USE_OBJECT_EQUALS&rule_sev=MAJOR&severity=MAJOR

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/FedoraResourceImpl.java [hunk 2]
```diff
@@ -192,6 +192,11 @@ protected FedoraResource doBackward(final FedoraResource fedoraResource) {
             }
             return fedoraResource;
         }
+
+        @Override
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/FedoraResourceImpl.java]: Not supporting `hashCode()` seems like an a sign of a more fundamental issue.
But any ways, the related Sonar issue seems to be gone: http://sonar.fcrepo.org/drilldown/issues/1?&rule=findbugs%3AEQ_COMPARETO_USE_OBJECT_EQUALS&rule_sev=MAJOR&severity=MAJOR

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/identifiers/HashConverter.java [hunk 3]
```diff
@@ -42,4 +42,9 @@ protected String doBackward(final String internalId) {
         }
         return internalId;
     }
+
+    @Override
+    public int hashCode() {
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/identifiers/HashConverter.java]: Not supporting `hashCode()` seems like an a sign of a more fundamental issue.
But any ways, the related Sonar issue seems to be gone: http://sonar.fcrepo.org/drilldown/issues/1?&rule=findbugs%3AEQ_COMPARETO_USE_OBJECT_EQUALS&rule_sev=MAJOR&severity=MAJOR

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/identifiers/HierarchyConverter.java [hunk 4]
```diff
@@ -162,4 +162,9 @@ public void setPrefix(final String p) {
         this.prefix = p;
     }
 
+    @Override
+    public int hashCode() {
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/identifiers/HierarchyConverter.java]: Not supporting `hashCode()` seems like an a sign of a more fundamental issue.
But any ways, the related Sonar issue seems to be gone: http://sonar.fcrepo.org/drilldown/issues/1?&rule=findbugs%3AEQ_COMPARETO_USE_OBJECT_EQUALS&rule_sev=MAJOR&severity=MAJOR

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/identifiers/NamespaceConverter.java [hunk 5]
```diff
@@ -67,4 +67,10 @@ protected String doBackward(final String b) {
         }
         return result;
     }
+
+    @Override
+    public int hashCode() {
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/identifiers/NamespaceConverter.java]: Not supporting `hashCode()` seems like an a sign of a more fundamental issue.
But any ways, the related Sonar issue seems to be gone: http://sonar.fcrepo.org/drilldown/issues/1?&rule=findbugs%3AEQ_COMPARETO_USE_OBJECT_EQUALS&rule_sev=MAJOR&severity=MAJOR

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/identifiers/NodeResourceConverter.java [hunk 6]
```diff
@@ -63,4 +63,9 @@ protected FedoraResource doForward(final Node node) {
     protected Node doBackward(final FedoraResource resource) {
         return resource.getNode();
     }
+
+    @Override
+    public int hashCode() {
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/identifiers/NodeResourceConverter.java]: Not supporting `hashCode()` seems like an a sign of a more fundamental issue.
But any ways, the related Sonar issue seems to be gone: http://sonar.fcrepo.org/drilldown/issues/1?&rule=findbugs%3AEQ_COMPARETO_USE_OBJECT_EQUALS&rule_sev=MAJOR&severity=MAJOR

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/converters/PropertyConverter.java [hunk 7]
```diff
@@ -152,4 +152,9 @@ public static String getPropertyNameFromPredicate(final NamespaceRegistry namesp
 
     }
 
+    @Override
+    public int hashCode() {
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/converters/PropertyConverter.java]: Not supporting `hashCode()` seems like an a sign of a more fundamental issue.
But any ways, the related Sonar issue seems to be gone: http://sonar.fcrepo.org/drilldown/issues/1?&rule=findbugs%3AEQ_COMPARETO_USE_OBJECT_EQUALS&rule_sev=MAJOR&severity=MAJOR

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/converters/ValueConverter.java [hunk 8]
```diff
@@ -268,4 +268,9 @@ public boolean isResource() {
             return hasDatatypeUri() && datatypeUri.equals("URI");
         }
     }
+
+    @Override
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/converters/ValueConverter.java]: Not supporting `hashCode()` seems like an a sign of a more fundamental issue.
But any ways, the related Sonar issue seems to be gone: http://sonar.fcrepo.org/drilldown/issues/1?&rule=findbugs%3AEQ_COMPARETO_USE_OBJECT_EQUALS&rule_sev=MAJOR&severity=MAJOR

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/DefaultIdentifierTranslator.java [hunk 9]
```diff
@@ -139,4 +139,9 @@ public String asString(final Resource subject) {
         return absPath;
     }
 
+    @Override
+    public int hashCode() {
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/DefaultIdentifierTranslator.java]: Not supporting `hashCode()` seems like an a sign of a more fundamental issue.
But any ways, the related Sonar issue seems to be gone: http://sonar.fcrepo.org/drilldown/issues/1?&rule=findbugs%3AEQ_COMPARETO_USE_OBJECT_EQUALS&rule_sev=MAJOR&severity=MAJOR

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpResourceConverter.java (hunk 1 above)
Comment (osmandin) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpResourceConverter.java]: Meaning, leave out hashcode?

Comment (awoods) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpResourceConverter.java]: Yes, it is not clear to me that overriding `hashCode()` is something we want to do.
Additionally, I am not seeing an indication in Sonar that would suggest we override `hashCode()`.

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4/fcrepo4/commit/58874be0f983c74e8a99040cce220908fca317f9

**Pull request closed without merging**